### ISSUE TITLE
Kriging numerical issues

### ISSUE THREAD

**Issue by LDAP** [SMTorg/smt#291]
I run a bunch of benchmarks and got suspicious when observing NRMSE rising with more samples.
It seems that Kriging is either broken with dim > 1or that there are serious numerical issues.

Python 3.9.1
numpy 1.20.1, scipy 1.6.0, scikit-learn 0.24.1

Real function:
![realfun](https://user-images.githubusercontent.com/8215084/107644919-5aca1300-6c78-11eb-920d-a32d1967ec09.png)

```python
# ...create test function...

sm = FullFactorial(xlimits=xlimits)
samples = sm(64)

kernel = RBF(1, 1, (0,1), (0,5))
gp = GP(kernel, print_debug=True)

gp.set_training_inputs(samples, realfun(samples))
gp.train()

krg = KRG()
krg.set_training_values(samples, realfun(samples))
krg.train()

# ...plot results...
```

SMT Kriging result (default settings):
![smt_krg](https://user-images.githubusercontent.com/8215084/107645007-7503f100-6c78-11eb-8aa1-16e25b20ba63.png)

My own Gaussian Process implementation:
![gp](https://user-images.githubusercontent.com/8215084/107645067-88af5780-6c78-11eb-844c-255c8b117803.png)

**Comment by relf** [comment]
Thanks for reporting. Which SMT version are you using? Could you provide your training data?

**Comment by LDAP** [comment]
SMT Commit: 257a464ced4ec3b2719d924681442cc719fa9f41

[data.zip](https://github.com/SMTorg/smt/files/5966236/data.zip)

**Comment by LDAP** [comment]
I wanted to add that I observe huge memory usage spikes when using KRG. Using my own implementation the memory usage is nearly constant.

**Comment by Paul-Saves** [comment]
I think that I have found the problem.
It comes from theta0 which was set to 1e-2 by default.
Therefore converging towards 1e-6 (lower bound) instead of 4. 
We recently modified the hyperparameters bounds and normalization but theta0 was not changed.

Before : 
krg = KRG(print_global=False,theta0=[1e-2])
![image](https://user-images.githubusercontent.com/56302141/107768785-49940b80-6d37-11eb-989b-c19712955a42.png)

After : 
krg = KRG(print_global=False,theta0=[1])
![image](https://user-images.githubusercontent.com/56302141/107768832-5b75ae80-6d37-11eb-81a1-fea5279bdf82.png)

see PR #292 .

**Comment by Paul-Saves** [comment]
Thank you for finding this problem :) 
To be sure it is corrected, can you try : 

```
krg = KRG( **theta0=[1.0]** )
krg.set_training_values(samples, realfun(samples))
krg.train()
```
My thanks !

**Comment by LDAP** [comment]
Using theta0=1 does yield the correct results for this data. Can you briefly explain how normalization is done and how it affects the hyperparameters? So do I understand correctly, that KrigingBased does only local optimization starting from theta0 (and noise0)?

How does this implementation perform, when there is a local minimum at theta=0.7 and the global minimum is at theta=0.02? Have I to adjust theta0 every time according to my data? My own implementation tries to optimize the parameters globally.

With another example, I did observe using eval_noise=True with noise0=[0] does not optimize noise correctly, using noise0=[1] does work as expected:

noisy sine wave (sigma=0.1), noise0=[0], theta0=[1], eval_noise=True
![grafik](https://user-images.githubusercontent.com/8215084/107772252-4d765c80-6d3c-11eb-9165-b0bb84ded46e.png)

noisy sine wave (sigma=0.1), noise0=[1], theta0=[1], eval_noise=True
![grafik](https://user-images.githubusercontent.com/8215084/107772330-654de080-6d3c-11eb-8fb3-e1f12e1a6101.png)

**Comment by Paul-Saves** [comment]
We normalize data before computing the hyperparameters, which are "auto-correlation length". 
For the optimization of the hyperparameters, we use scipy.optimize COBYLA and it performs a local optimization.

I tried to add a small multistart, it is not perfect but better than before. I agree with you, it is not normal that one should set the starting point every time.

I used your noisy sine example : 
Before : 
![image](https://user-images.githubusercontent.com/56302141/107791729-90442e80-6d54-11eb-9ff4-bad6029e1f48.png)

After :
![image](https://user-images.githubusercontent.com/56302141/107788199-6db01680-6d50-11eb-8bd2-217a0a0f3f0b.png)

**Comment by relf** [comment]
@LDAP, #293 thanks to @Paul-Saves is merged, it should have improved the situation, could you give a feedback whether it works for you?

**Comment by LDAP** [comment]
Yes it performs much better. Many thanks @Paul-Saves!

**Comment by Paul-Saves** [comment]
Thanks you too for finding out the issue and contributing to SMT :)
